src/decoder.rs

```diff
@@ -124,6 +127,13 @@ impl<R: Read> Decoder<R> {
         }
     }
 
+    /// Returns raw exif data, starting at the TIFF header, if the image contains any.
+    ///
+    /// The returned value will be `None` until a call to `decode` has returned `Ok`.    
+    pub fn exif_data(&self) -> Option<Vec<u8>> {
+        self.exif_data.as_ref().map(|v| v.clone())
```

Comment:
Wouldn't `Option<&[u8]>` make more sense? `self.exif_data.as_ref()` only. That way you don't need to copy the content the whole time. If people need an owned type, they can simply convert the slice to a vec themselves 🤔

Reply:
You mean `as_deref()` right? So option 3:
```rust
pub fn exif_data(&self) -> Option<&[u8]> {
        self.exif_data.as_deref()
    }
```
Sorry for the rookie mistakes. 😄

Reply:
Nevermind, different rust version.

Reply:
A `match` works as well:
```rust
match &self.exif_data {
    None => None,
    Some(data) => Some(data.as_slice()),
}
```

Reply:
I initially went with 
```rust 
self.exif_data.as_ref().map(|v| v.as_ref())
```
but `as_slice()` seems a bit more obvious.

Reply:
Is there a MSRV? Because if not, that could be written using `as_deref`, which is way cleaner

Reply:
There are CI checks that run on 1.34.2, they fail when using `as_deref`. Not sure if there is a policy though.

Reply:
I believe that 1.34.2 is our MSRV. We occasionally bump that, but I don't think it is worth it in this case